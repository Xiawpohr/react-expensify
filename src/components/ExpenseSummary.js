import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import getVisibleExpenses from '../selectors/getVisibleExpenses.js'
import getExpensesTotal from '../selectors/getExpensesTotal.js'

export const ExpenseSummary = (props) => {
  const expenseWord = props.count === 1 ? 'expense' : 'expenses'
  const formattedTotal = numeral(props.total / 100).format('$0,0.00')
  return (
    <div>
      <h2>Viewing {props.count} {expenseWord} totalling {formattedTotal}</h2>
    </div>
  )
}

const mapStateToProps = state => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
  const total = getExpensesTotal(visibleExpenses)
  return {
    count: visibleExpenses.length,
    total
  }
}

export default connect(mapStateToProps)(ExpenseSummary)
