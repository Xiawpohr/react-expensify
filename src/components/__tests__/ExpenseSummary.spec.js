import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../ExpenseSummary.js'

test('should correctly render Expenses Summary without expenses', () => {
  const count = 1
  const total = 123
  const wrapper = shallow(<ExpenseSummary count={count} total={total} />)
  expect(wrapper).toMatchSnapshot()
})

test('should correctly render Expenses Summary with multiple expenses', () => {
  const count = 10
  const total = 12345.67
  const wrapper = shallow(<ExpenseSummary count={count} total={total} />)
  expect(wrapper).toMatchSnapshot()
})
