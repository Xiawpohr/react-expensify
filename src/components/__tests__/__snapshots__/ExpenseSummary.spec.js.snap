// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expenses Summary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    10
     
    expenses
     totalling 
    $123.46
  </h2>
</div>
`;

exports[`should correctly render Expenses Summary without expenses 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h2>
</div>
`;
